x/nexus/keeper/lockable_asset.go

```diff
@@ -31,14 +31,14 @@ type lockableAsset struct {
 func newLockableAsset(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, bank types.BankKeeper, coin sdk.Coin) (lockableAsset, error) {
 	denom := coin.GetDenom()
```

Comment:
Perhaps `Asset.Validate` should restrict assets of type `ibc/{hash}` to avoid inadvertently creating nexus registered assets that also look like ICS20 coins. It's a permissioned cmd, but the check makes this explicit and easier to understand? If we keep the shared constructor for the two representations, this would make it a bit safer